azavea/ckanext-odp_theme
# Remove 'unpublished feedback' feature

### Overview

Removes the parts of this extension related to adding the ability to leave feedback on unpublished datasets:
- the UnpublishedFeedback controller and model
- templates, javascript, and styles that implement the feature
- the related additions and overrides from `plugin.py`
- customizations for making the "published" custom dataset field
  visible and filterable
- the custom `package/read.html` and `snippets/facet_list.html`
  templates because all of the modifications were related to these features

Note: you may want to delete the `ckanext_unpublished_feedback` and `ckanext_unpublished_comments` tables from your database, if they exist.

UPDATE: While I was doing destruction, I also removed the ` ckanext_odp_theme.egg-info` directory and added it to .gitignore.

### Notes

This feature was launched but not adopted--the ability to filter by published status and leave feedback on unpublished datasets was operational, but the only unpublished datasets were dummy/test ones.

Even if we wanted to keep this feature, it should be split out into an extension of its own rather than combined with `odp_theme`, since it's a fairly self-contained piece of functionality and not particularly related to theming the site.

### Testing

- Comment the `ckanext-odp_theme` role back in in [app.yml](https://github.com/azavea/opendataphilly-ckan/blob/develop/deployment/ansible/app.yml).  If you provisioned your `app` instance without `odp_theme` before you can just reprovision. Otherwise, destroy and rebuild.
- Confirm that there's no "Published" section in the sidebar filters on http://localhost:8025/dataset
- Confirm that http://localhost:8025/dataset/unpublished-test doesn't have a "Feedback" form

Resolves https://github.com/azavea/urban-apps/issues/159
